### PR TITLE
Docs for installing python API in M1

### PR DESCRIPTION
**Pull request by yoni206**
Adds documentation instructions for the python API on M1 MacOS.

### CODE REVIEW

**Comment by yoni206** [comment]
@barrettcw FYI.

File: docs/api/python/python.rst [hunk 1]
```diff
@@ -28,11 +28,33 @@ If you would like a more feature-complete---yet verbose---python API, with the a
...
 
 The base and pythonic Python API can be installed via `pip` as follows:
 
 .. code:: bash
 
   pip install cvc5
+
+
+Installation (MacOS with Apple Silicon)
+------------
```
Comment (4tXJ7f) [inline docs/api/python/python.rst:42]:
```suggestion
Installation (ARM64 variants of Linux and macOS)
------------
```

File: docs/api/python/python.rst [hunk 2]
```diff
@@ -28,11 +28,33 @@ If you would like a more feature-complete---yet verbose---python API, with the a
 You can compare examples using the two APIs by visiting the :doc:`examples page <../../examples/quickstart>`.
 
 
-Installation
+Installation (Except MacOS with Apple Silicon)
```
Comment (4tXJ7f) [inline docs/api/python/python.rst:31]:
```suggestion
Installation (x86-64 variants of Linux and macOS)
```
The instructions don't work for Windows or ARM versions of Linux, so it is better to explicitly list what's supported.

File: docs/api/python/python.rst [hunk 3]
```diff
@@ -28,11 +28,33 @@ If you would like a more feature-complete---yet verbose---python API, with the a
...
 
 .. code:: bash
 
   pip install cvc5
+
+
+Installation (MacOS with Apple Silicon)
+------------
+
+For M1 Macs, the base and pythonic Python API can be installed from source as follows:
```
Comment (4tXJ7f) [inline docs/api/python/python.rst:44]:
```suggestion
For ARM64-based machines (including Apple computers with M1 and M2 chips), the base and the pythonic Python API can be installed from source as follows:
```

**Review by 4tXJ7f — CHANGES_REQUESTED** [review]
Thank you for doing this. Some minor suggestions.

**Comment by yoni206** [comment]
Thanks for your suggestions! I have committed them.

**Review by 4tXJ7f — APPROVED** [review]
LGTM

**Pull request merged**